Engine_Revit_UI/Modify/SetParameter.cs

```diff
@@ -133,8 +133,29 @@ public static bool SetParameter(this Element element, BuiltInParameter builtInPa
 
         public static bool SetParameter(this Parameter parameter, object value, Document document = null)
         {
-            if (parameter == null || parameter.IsReadOnly)
-                return false;
+            // Workset parameters
+            // Filter for worksets
+            if (parameter.Id.IntegerValue == (int)BuiltInParameter.ELEM_PARTITION_PARAM)
```

Comment:
If `parameter` is `null` on the entry point then this will throw a `NullReferenceException`.

Reply:
Null check moved up as suggested